### PR TITLE
Add option to set rebuild action

### LINKED ISSUE #4: Support `boot`, `test` and `dry-activate` actions
Maybe also add ability to pass flags like `--use-remote-sudo` while at it.
Relevant code: https://github.com/pinpox/lollypops/blob/ad2d9b9504fefbcabfe92c734d80f2f3a5e3b69e/flake.nix#L95

### PR DESCRIPTION
**Pull request by pinpox**
fixes #4

### CODE REVIEW

**Comment by pinpox** [comment]
> It's kinda weird to set rebuild action in the config, shouldn't it be an argument on the command line?

I can also make it an argument, but the syntax is not very pretty. Since I'm using go-task we need to use `--` to separate arguments passed to the commands as shown here: https://taskfile.dev/usage/#forwarding-cli-arguments-to-commands

Together with nix this would result in:

`nix run '.' -- hostname -- action`

A different approach would be to use environment variables for the action. E.g. setting the nixos-rebuild action to `$LP_NIXOS_ACTION` and defaulting to `switch`.  
Not sure what way to go here yet, maybe there is a third, better option aswell.

**Comment by misuzu** [comment]
> A different approach would be to use environment variables for the action. E.g. setting the nixos-rebuild action to `$LP_NIXOS_ACTION` and defaulting to `switch`.

This looks like a better approach.

File: flake.nix [hunk 1]
```diff
@@ -56,6 +56,7 @@
                       vars = with hostConfig.config.lollypops; {
                         REMOTE_USER = deployment.user;
                         REMOTE_HOST = deployment.host;
+                        REBUILD_ACTION = deployment.rebuild-action;
```
Comment (misuzu) [inline flake.nix:59]: Having `rebuild-action` in configuration feels weird tbh.
I think it's enough to do something like this: `REBUILD_ACTION = "{{default \"switch\" .REBUILD_ACTION}}";`
This should allow running `REBUILD_ACTION=dry-activate nix run . -- <hostname>` if needed.

Comment (pinpox) [reply flake.nix]: Agreed, this is a good way to do it.

**Pull request merged**